### PR TITLE
tools/unitctl: implement application subcommand

### PR DESCRIPTION
**Pull request by avahahn**
This MR addresses #1280 and #1279 by providing a subcommand `unitctl app` that provides two options:

* `unitctl app list`: returns a list of running applications
```console
$ unitctl app list -t json-pretty
{
  "wasm": {
    "type": "wasm-wasi-component",
    "component": "/www/wasmapp-proxy-component.wasm"
  }
}
```

* `unitctl app <NAME> reload`: reloads an application by name
```console
$ unitctl app reload wasm -t json-pretty
{
  "success": "Ok"
}
```

This MR also adds missing wasm and wasi application schemas to the OpenAPI spec.

### CODE REVIEW

**Review by javorszky — APPROVED** [review]
Two questions, but both are non-blocking. Looks good.

Installation on a mac M series chip needs openjdk rather than java from java.com (latest is JRE 8 v 411 at the time of writing, only supports class file version 52.0) as openapi files have class file version 55.0

File: tools/unitctl/GNUmakefile [hunk 1]
```diff
@@ -137,7 +137,7 @@ openapi-clean: ## Clean up generated OpenAPI files
 	$Q find "$(CURDIR)/unit-openapi/src/apis" \
 		! -name 'error.rs' -type f -exec rm -f {} +
 	$Q $(info $(M) cleaning up generated OpenAPI models code)
-	$Q rm -rf "$(CURDIR)/unit-openapi/src/models/*"
+	$Q rm -rf "$(CURDIR)/unit-openapi/src/models"
```
[the remark this answers is not in the document]
Comment (javorszky) [reply tools/unitctl/GNUmakefile:140]: <on a zoom call> yes

File: tools/unitctl/README.md [hunk 2]
```diff
@@ -91,6 +91,26 @@ To the subcommand `unitctl instances new` the user must provide three things:
 
 After deployment the user will have one Unit container running on the host network.
 
+### Lists active applications and provides means to restart them
+Listing applications:
+```
+$ unitctl app list -t json-pretty
```
[the remark this answers is not in the document]
Comment (javorszky) [reply tools/unitctl/README.md:97]: command help content already has this info in it

File: tools/unitctl/unitctl/src/unitctl.rs [hunk 3]
```diff
@@ -114,6 +114,8 @@ pub(crate) enum Commands {
         )]
         output_format: OutputFormat,
     },
+    #[command(about = "List all configured UNIT applications")]
```
Comment (callahad) [inline tools/unitctl/unitctl/src/unitctl.rs:117]:
```suggestion
    #[command(about = "List all configured Unit applications")]
```

...and we should probably check that we're consistent elsewhere in unitctl.

File: tools/unitctl/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (callahad) [reply tools/unitctl/README.md:97]: `json-pretty` is documented as the default; let's omit it here (and below) to keep the focus on the application itself.

```suggestion
$ unitctl app list
```

File: tools/unitctl/unitctl/src/cmd/instances.rs [hunk 4]
```diff
@@ -22,66 +22,69 @@ pub(crate) async fn cmd(args: InstanceArgs) -> Result<(), UnitctlError> {
...
                         if cfg!(target_os = "macos") {
-                            return Err(UnitctlError::ControlSocketError{
+                            return Err(UnitctlError::ControlSocketError {
                                 kind: ControlSocketErrorKind::General,
-                                message: format!("Docker on OSX will break unix sockets mounted {} {}",
-                                                 "in containers, see the following link for more information",
-                                                 "https://github.com/docker/for-mac/issues/483"),
-                            })
+                                message: format!(
+                                    "Docker on OSX will break unix sockets mounted {} {}",
```
Comment (callahad) [inline tools/unitctl/unitctl/src/cmd/instances.rs:45]:
```suggestion
                                    "Docker on macOS will break unix sockets mounted {} {}",
```

(Sorry)

**Review by callahad — APPROVED** [review]
Some nits around prose, but otherwise lgtm.

I _really_ wish the whitespace / rustfmt changes were broken into their own commits. Next time? 🙏

File: tools/unitctl/unitctl/src/unitctl.rs (hunk 3 above)
Comment (ac000) [reply tools/unitctl/unitctl/src/unitctl.rs:117]: I wouldn't say it's a suggestion!

Unit is _never_ spelled UNIT (it's not an acronym)

File: tools/unitctl/unitctl/src/cmd/instances.rs (hunk 4 above)
Comment (ac000) [reply tools/unitctl/unitctl/src/cmd/instances.rs:45]: "unix sockets" should probably be "UNIX domain sockets"

**Pull request merged**